File: contains_duplicate/contains_duplicate.py
# My Answer
def containsDuplicate(self, nums: List[int]) -> bool:
    mySet = set(nums)
    if len(mySet) == len(nums):
        return False
    else:
        return True


# Using Counter
from collections import Counter


def containsDuplicate(self, nums: List[int]) -> bool:
    for k, v in Counter(nums).items():
        if v > 1:
            return True
    return False

File: leet_code_algorithms/contains_duplicate/contains_duplicate.ts
// My Answer
function containsDuplicate(nums: number[]): boolean {
  nums.sort();
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == nums[i + 1]) {
      return true;
    }
  }
  return false;
}

// Using Set
function containsDuplicateTwo(nums: number[]): boolean {
  const set = new Set();
  for (const num of nums) {
    if (set.has(num)) {
      return true;
    }
    set.add(num);
  }
  return false;
}

function containsDuplicateThree(nums: number[]): boolean {
  return Array.from(new Set(nums)).length != nums.length;
}

//using Map
function containsDuplicateMap(nums: number[]): boolean {
  const numCount = new Map<number, number>();
  for (const num of nums) {
    if (numCount.has(num)) {
      return true;
    }
    numCount.set(num, 1);
  }
  return false;
}
